android-sdk/src/main/java/com/mobilecoin/lib/exceptions/InvalidTransactionException.java

```diff
@@ -2,10 +2,12 @@
 
 package com.mobilecoin.lib.exceptions;
 
-import androidx.annotation.Nullable;
-
 public final class InvalidTransactionException extends MobileCoinException {
-    public InvalidTransactionException(@Nullable String message) {
+    public final String message;
```

Comment:
my comment at https://github.com/mobilecoinofficial/mobilecoin_flutter_plugin/pull/13 presumes adding an error code enum into the exception itself. So that we don't have to compare strings, the usage would look something like this: 
```
catch (InvalidTransactionException ex) { 
    switch (ex.getResultCode()) {
        case TransactionSubmissionStatus.ContainsSpentImages: 
        ....
    }
}

```
The constructor would look like this:
InvalidTransactionException(@NonNull TransactionSubmissionStatus status, @Nullable Throwable throwable)
where `TransactionSubmissionStatus` is a new class

See [NetworkException](https://github.com/mobilecoinofficial/android-sdk/blob/master/android-sdk/src/main/java/com/mobilecoin/lib/exceptions/NetworkException.java) for an example

Reply:
Would this be a breaking change, then? Or just an additional constructor and deprecate the other one? 

[It doesn't look like Signal uses the old constructor](https://github.com/signalapp/Signal-Android/search?q=InvalidTransactionException), even though it's packaged as public.